dask/dataframe/io/tests/test_csv.py

```diff
@@ -208,6 +209,25 @@ def test_text_blocks_to_pandas_kwargs(reader, files):
     assert (result.columns == df.columns).all()
 
 
+@csv_and_table
+def test_text_blocks_to_pandas_has_different_task_names_based_on_blocksize(
+    reader, files
+):
+    blocks = [[files[k]] for k in sorted(files)]
+    kwargs = {}
+    head = pandas_read_text(reader, files["2014-01-01.csv"], b"", {})
+    header = files["2014-01-01.csv"].split(b"\n")[0] + b"\n"
+
+    df = text_blocks_to_pandas(reader, blocks, header, head, kwargs)
+    df_names = {key[0] for key in df.dask.keys()}
+
+    values = text_blocks_to_pandas(
+        reader, blocks, header, head, kwargs, blocksize="30B"
+    )
+    value_names = {key[0] for key in values.dask.keys()}
+    assert value_names != df_names
```

Comment:
This works, and I'm happy to see this merged in.  However, if we wanted to use more public API and make things a bit easier for future maintainers we might do the following instead:

```python
def test_different_names():
    with filetext(text) as fn:
        a = dd.read_csv(fn, blocksize="10kB")
        b = dd.read_csv(fn, blocksize="20kB") 
        assert a.name != b.name
```

Reply:
It's the goldilocks of tests! I'm happy to change it to that.